src/init/features/dataconnect/sdk.ts

```diff
@@ -104,103 +101,89 @@ async function askQuestions(setup: Setup, config: Config): Promise<SDKInfo> {
     choices: connectorChoices,
   });
 
-  const newConnectorYaml = JSON.parse(JSON.stringify(connectorInfo.connectorYaml)) as ConnectorYaml;
-  if (!newConnectorYaml.generate) {
-    newConnectorYaml.generate = {};
+  const connectorYaml = JSON.parse(JSON.stringify(connectorInfo.connectorYaml)) as ConnectorYaml;
+  const newConnectorYaml = generateSdkYaml(
+    targetPlatform,
+    connectorYaml,
+    connectorInfo.directory,
+    appDir,
+  );
+
+  const shouldGenerate = !!(
+    setup.projectId &&
+    (await confirm({
+      message: "Would you like to generate SDK code now?",
+      default: true,
+    }))
+  );
+  // TODO: Prompt user about adding generated paths to .gitignore
+  const connectorYamlContents = yaml.stringify(newConnectorYaml);
+  connectorInfo.connectorYaml = newConnectorYaml;
+  const displayIOSWarning = targetPlatform === Platform.IOS;
+  return { connectorYamlContents, connectorInfo, shouldGenerate, displayIOSWarning };
+}
+
+export function generateSdkYaml(
+  targetPlatform: Platform,
+  connectorYaml: ConnectorYaml,
+  connectorDir: string,
+  appDir: string,
+): ConnectorYaml {
+  if (!connectorYaml.generate) {
+    connectorYaml.generate = {};
   }
 
-  let displayIOSWarning = false;
   if (targetPlatform === Platform.IOS) {
-    const outputDir =
-      newConnectorYaml.generate.swiftSdk?.outputDir ||
-      path.relative(connectorInfo.directory, path.join(appDir, `generated/swift`));
-    const pkg =
-      newConnectorYaml.generate.swiftSdk?.package ??
-      upperFirst(camelCase(newConnectorYaml.connectorId));
-    const swiftSdk = { outputDir, package: pkg };
-    newConnectorYaml.generate.swiftSdk = swiftSdk;
-    displayIOSWarning = true;
+    const swiftSdk = {
+      outputDir: path.relative(connectorDir, path.join(appDir, `dataconnect-generated/swift`)),
+      package: upperFirst(camelCase(connectorYaml.connectorId)) + "Connector",
+    };
+    connectorYaml.generate.swiftSdk = swiftSdk;
   }
 
   if (targetPlatform === Platform.WEB) {
-    const outputDir =
-      newConnectorYaml.generate.javascriptSdk?.outputDir ||
-      path.relative(
-        connectorInfo.directory,
-        path.join(appDir, `generated/javascript/${newConnectorYaml.connectorId}`),
-      );
-    const pkg =
-      newConnectorYaml.generate.javascriptSdk?.package ??
-      `@firebasegen/${connectorInfo.connectorYaml.connectorId}`;
-
+    const pkg = `${connectorYaml.connectorId}-connector`;
     const javascriptSdk: JavascriptSDK = {
-      outputDir,
-      package: pkg,
-      packageJsonDir: newConnectorYaml.generate.javascriptSdk?.packageJsonDir,
+      outputDir: path.relative(connectorDir, path.join(appDir, `dataconnect-generated/js/${pkg}`)),
```

Comment:
This seems different from the swiftSdk `outputDir`

Reply:
Aim for the same pattern `dataconnect-generated/{js,swift,kotlin}/${package}` for all platforms.

`${package}` is part of `outputDir` of `JS` and `Dart`, but implicitly included for `Swift` and `Kotlin`.

Reply:
Why would it be a breaking change? Isn't this just for when the user hasn't created a generated SDK for the platform yet?